SonarQube.TeamBuild.PreProcessor/Roslyn/PluginResourceCache.cs

```diff
@@ -0,0 +1,111 @@
+﻿/*
+ * SonarQube Scanner for MSBuild
+ * Copyright (C) 2015-2017 SonarSource SA and Microsoft Corporation
+ * mailto: contact AT sonarsource DOT com
+ *
+ * Licensed under the MIT License.
+ * See LICENSE file in the project root for full license information.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using Newtonsoft.Json;
+
+namespace SonarQube.TeamBuild.PreProcessor.Roslyn
+{
+    class SubdirIndex
+    {
+        // global locking, to ensure synchronized access to index file by multiple processes
+        private readonly EventWaitHandle waitHandle = new EventWaitHandle(true, EventResetMode.AutoReset, "90CD3CFF-A12C-4013-A44A-199B8C26818B");
```

Comment:
The _waitHandle_ is disposable and should be disposed; however, the owning process is fairly short-lived and it should be cleaned up automatically when the process terminates.